zip-0227.rst

```diff
@@ -1,8 +1,359 @@
+
+
+NOTE that the commitment is not included in the `IssuanceAction` itsefl. As explained below, it is later computed by the validators and added to the `NoteCommitmentTree`.
+
+Concrete applications
+---------------------
+
+**Bridging Assets**
+Issuers can wrap assets defined in other chains and issue them at once in a single transaction. The specifics of the bridge are not implemented in the protocol, but there are several ways we conceive issuers can build these bridges (at least centralized bridges):
+- First, the issuance and burn mechanism can be used in conjunction to determine the 
+
+**Asset Features**
+- By using the `finalize` boolean and the burning mechanism defined in [#zip-0226]_, issuers can control the supply production of any asset associated to their issuer keys. For example,
+    - by setting `finalize = 1` from the first issuance action for that asset type, the issuer is in essence creating a one-time issuance transaction. This is useful when the max supply is capped from the beginning and the distribution is known in advance. All tokens are issued at once and distributed as needed.
+- Issuers can also stop the existing supply production of any asset associated to their issuer keys. This could be done by
+    - issuing a last set of tokens of that specific `AssetId`, while at the same time setting `finalize = 1`, or by
+    - issuing a transaction with a single note in the issuance action pertaining to that `AssetId`, where the note will contain a `value = 0`. This can be used for application-specific purposes (NFT collections) or for security purposes to revoke the asset issuance (see Security and Privacy Considerations).
+- Furthermore, NFT issuance is enabled by issuing in a single bundle several issuance actions, where each `AssetId` corresponds to `value = 1` at the fundamental unit level. Issuers and users should make sure that `finalize = 1` for each of the actions in this scenario.
+
+Consensus Rule Changes
+----------------------
+
+For the IssueBundle,
+- Verify the RedPallas-based issuance authorization signature on `SIGHASH`, `authorization` is valid, based on `authorization.VerifySig(ik, SIGHASH)`
+
+For each `IssueAction` in `IssueBundle`:
+- check that `asset_desc` is a string with `0 < byte size <= 512`
+- retrieve `AssetId` from the first note in the sequence and check that `AssetId = GroupHash^{\mathbb{P}}(ik || asset_desc)` is properly derived
+- check that the `AssetId` does not exist in the `previously_finalized` set in the global state
+- check that every note in the `IssueAction` contains the same `AssetId` and also that every note is of type `Note` and is properly constructed as :math:`note = (\mathsf{g_d, pk_d, v, \rho, \psi, AssetId})`
+
+If all of the above checks pass, do the following:
+- For each note, compute the note commitment as :math:`cm = \mathsf{NoteCommit^{OrchardZSA}_{rcm}(repr_{\mathbb{P}}(g_d), repr_{\mathbb{P}}(pk_d), v, \rho, \psi, AssetId)}` and
```

Comment:
This detail should be wrapped up as a function that can be invoked here, instead of duplication and possible divergence

Reply:
Done.